src/openms/source/ANALYSIS/ID/IDRipper.cpp

```diff
@@ -160,8 +187,84 @@ namespace OpenMS
         peptide_idents.push_back(*pep_it);
 
         //create and insert new map entry
-        ripped.insert(make_pair(file_, make_pair(protein_idents, peptide_idents)));
+        ripped.insert(make_pair(idx_pair, make_pair(protein_idents, peptide_idents)));
```

Comment:
inefficient. but maybe ok, if this is not a performance bottleneck.
better alternative:
`.emplace()` with `piecewise_construct` and std::move

Reply:
This happens once per output file, which are usually in the order of < 5.